src/Core/Notifications/INotificationService.cs

```diff
@@ -18,10 +18,89 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
+using System.Linq;
+using SonarLint.VisualStudio.Core.InfoBar;
+
 namespace SonarLint.VisualStudio.Core.Notifications
 {
-    public interface INotificationService
+    public interface INotificationService : IDisposable
     {
         void ShowNotification(INotification notification);
     }
+    
+    public sealed class NotificationService : INotificationService
+    {
+        /// <summary>
+        /// Taken from "ToolWindowGuids80.ErrorList"
+        /// </summary>
+        internal static readonly Guid ErrorListToolWindowGuid = new Guid("D78612C7-9962-4B83-95D9-268046DAD23A");
+
+        private readonly IInfoBarManager infoBarManager;
+        private readonly IThreadHandling threadHandling;
+
+        private Tuple<IInfoBar, INotification> activeNotification;
+
+        public NotificationService(IInfoBarManager infoBarManager, IThreadHandling threadHandling)
+        {
+            this.infoBarManager = infoBarManager;
+            this.threadHandling = threadHandling;
+        }
+
+        public void ShowNotification(INotification notification)
+        {
+            if (notification == null)
+            {
+                throw new ArgumentNullException(nameof(notification));
+            }
+            // todo: check if already shown
+
+            threadHandling.RunOnUIThread(() =>
+            {
+                RemoveExistingInfoBar();
```

Comment:
This will need more work later too. There will probably be occasions when we want to display more than one (different) notification at the same time.